core/dataset.h

```diff
@@ -173,6 +173,20 @@ struct ColumnIndex {
     virtual uint64_t getColumnRowCount(const ColumnName & column) const;
 };
 
+/*****************************************************************************/
+/* ROW STREAM                                                                */
+/*****************************************************************************/
+
+struct RowStream {
+
+    virtual std::shared_ptr<RowStream> clone() const = 0;
+
+    /* set where the stream should start*/
```

Comment:
I wonder: could we change this to an interface like splitIntoManyStreams(numStreams)?  To avoid having to implement initAt, which must be slow?